build-tools/src/main/java/org/elasticsearch/gradle/testclusters/TestClusterConfiguration.java

```diff
@@ -0,0 +1,172 @@
+
+    void keystore(String key, File value);
+
+    void keystore(String key, File value, PropertyNormalization normalization);
+
+    void keystore(String key, FileSupplier valueSupplier);
+
+    void keystorePassword(String password);
+
+    void cliSetup(String binTool, CharSequence... args);
+
+    void setting(String key, String value);
+
+    void setting(String key, String value, PropertyNormalization normalization);
+
+    void setting(String key, Supplier<CharSequence> valueSupplier);
+
+    void setting(String key, Supplier<CharSequence> valueSupplier, PropertyNormalization normalization);
+
+    void systemProperty(String key, String value);
+
+    void systemProperty(String key, Supplier<CharSequence> valueSupplier);
+
+    void systemProperty(String key, Supplier<CharSequence> valueSupplier, PropertyNormalization normalization);
+
+    void environment(String key, String value);
+
+    void environment(String key, Supplier<CharSequence> valueSupplier);
+
+    void environment(String key, Supplier<CharSequence> valueSupplier, PropertyNormalization normalization);
+
+    void jvmArgs(String... values);
+
+    boolean isPreserveDataDir();
+
+    void setPreserveDataDir(boolean preserveDataDir);
+
+    void freeze();
+
+    void start();
+
+    void restart();
+
+    void extraConfigFile(String destination, File from);
+
+    void extraConfigFile(String destination, File from, PropertyNormalization normalization);
+
+    void extraJarFiles(FileCollection from);
+
+    void user(Map<String, String> userSpec);
+
+    String getHttpSocketURI();
+
+    String getTransportPortURI();
+
+    List<String> getAllHttpSocketURI();
+
+    List<String> getAllTransportPortURI();
+
+    void stop(boolean tailLogs);
+
+    void setNameCustomization(Function<String, String> nameSupplier);
+
+    default void waitForConditions(
+        LinkedHashMap<String, Predicate<TestClusterConfiguration>> waitConditions,
+        long startedAtMillis,
+        long nodeUpTimeout,
+        TimeUnit nodeUpTimeoutUnit,
+        TestClusterConfiguration context
+    ) {
+        Logger logger = Logging.getLogger(TestClusterConfiguration.class);
+        waitConditions.forEach((description, predicate) -> {
+            long thisConditionStartedAt = System.currentTimeMillis();
+            boolean conditionMet = false;
+            Throwable lastException = null;
+            while (System.currentTimeMillis() - startedAtMillis < TimeUnit.MILLISECONDS.convert(nodeUpTimeout, nodeUpTimeoutUnit)) {
+                if (context.isProcessAlive() == false) {
+                    throw new TestClustersException("process was found dead while waiting for " + description + ", " + this);
+                }
+
+                try {
+                    if (predicate.test(context)) {
+                        conditionMet = true;
+                        break;
+                    }
+                } catch (TestClustersException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)